lib/imagec/imagec.go

```diff
@@ -325,12 +325,16 @@ func (ic *ImageC) WriteImageBlob(image *ImageWithMeta, progressOutput progress.O
 // CreateImageConfig constructs the image metadata from layers that compose the image
 func (ic *ImageC) CreateImageConfig(images []*ImageWithMeta) (metadata.ImageConfig, error) {
 
-	if len(images) == 0 {
-		return metadata.ImageConfig{}, nil
+	imageLayer := images[0] // the layer that represents the actual image
+
+	// if we already have an imageID associated with this layerID, we don't need
+	// to calculate imageID and can just grab the image config from the cache
+	id := cache.RepositoryCache().GetImageID(imageLayer.ID)
+	if image, err := cache.ImageCache().GetImage(id); err == nil {
```

Comment:
I don't believe this can ever happen in any particular VIC codepath. However, I do not believe there currently to be any way of knowing if a user has deleted layer/image data from the datastore out-of-band. I don't think we are currently publishing any events when this happens, but I'm pinging @cgtexmex in case he knows for sure.